packages/react/src/useAction.ts

```diff
@@ -79,9 +77,43 @@ export const useAction = <
     transformedResult,
     useCallback(
       async (variables, context) => {
+        if (action.hasAmbiguousIdentifier) {
+          if (Object.keys(variables).some((key) => !action.paramOnlyVariables?.includes(key) && key !== action.modelApiIdentifier)) {
+            throw Error(`Invalid arguments found in variables. Did you mean to use ({ ${action.modelApiIdentifier}: { ... } })?`);
+          }
+        }
+
+        let newVariables: Exclude<F["variablesType"], null | undefined>;
+        if (action.hasCreateOrUpdateEffect) {
+          if (
+            action.modelApiIdentifier in variables &&
+            typeof variables[action.modelApiIdentifier] === "object" &&
+            variables[action.modelApiIdentifier] !== null
+          ) {
+            newVariables = variables;
+          } else {
+            newVariables = {
+              [action.modelApiIdentifier]: {},
+            } as Exclude<F["variablesType"], null | undefined>;
```

Comment:
This has to be casted? :(

Is it TypeScript not figuring it out, or do we have something typed incorrectly?

Reply:
I get `Type '{ [x: string]: {}; }' is not assignable to type 'Exclude<F["variablesType"], null | undefined>'.`